xclim/core/checks.py

```diff
@@ -1,461 +1,4 @@
-    nc : int
-      Number of consecutive missing values per month that should not be exceeded.
-    **indexer : {dim: indexer, }, optional
-      Time attribute and values over which to subset the array. For example, use season='DJF' to select winter values,
-      month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given, all values are
-      considered.
-
-    Returns
-    -------
-    out : DataArray
-      A boolean array set to True if period has missing values.
-    """
-    missing = MissingWMO(da, "M", **indexer)(nm=nm, nc=nc)
-    return missing.resample(time=freq).any()
-
-
-def missing_pct(da, freq, tolerance, **indexer):
-    r"""Return whether there are more missing days in the array than a given percentage.
-
-    Parameters
-    ----------
-    da : DataArray
-      Input array at daily frequency.
-    freq : str
-      Resampling frequency.
-    tolerance : float
-      Fraction of missing values that is tolerated.
-    **indexer : {dim: indexer, }, optional
-      Time attribute and values over which to subset the array. For example, use season='DJF' to select winter
-      values,
-      month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given, all values are
-      considered.
-
-    Returns
-    -------
-    out : DataArray
-      A boolean array set to True if period has missing values.
-    """
-    return MissingPct(da, freq, **indexer)(tolerance=tolerance)
-
-
-def at_least_n_valid(da, freq, n=1, **indexer):
-    r"""Return whether there are at least a given number of valid values.
-
-        Parameters
-        ----------
-        da : DataArray
-          Input array at daily frequency.
-        freq : str
-          Resampling frequency.
-        n : int
-          Minimum of valid values required.
-        **indexer : {dim: indexer, }, optional
-          Time attribute and values over which to subset the array. For example, use season='DJF' to select winter
-          values, month=1 to select January, or month=[6,7,8] to select summer months. If not indexer is given,
-          all values are considered.
-
-        Returns
-        -------
-        out : DataArray
-          A boolean array set to True if period has missing values.
-        """
-    return AtLeastNValid(da, freq, **indexer)(n=n)
-
-
-def missing_from_context(da, freq, **indexer):
-    """Return whether each element of the resampled da should be considered missing according
-    to the currently set options in `xclim.set_options`.
-
-    See `xclim.set_options` and `xclim.core.options.register_missing_method`.
-    """
-    name = OPTIONS[CHECK_MISSING]
-    cls = MISSING_METHODS[name]
-    opts = OPTIONS[MISSING_OPTIONS][name]
-
-    return cls(da, freq, **indexer)(**opts)
+# This file is kept for backward compatibility. It will be removed in a later version.
+from .cfchecks import *
+from .datachecks import *
+from .missing import *
```

Comment:
Maybe a solution would be to throw a DeprecationWarning directly here?

Reply:
Ok, I'll open another ticket for this.